Setup/Patch/Data/RebuildReplicasPatch.php

```diff
@@ -52,7 +52,7 @@ public function getAliases(): array
      */
     public function apply(): PatchInterface
     {
-        if (!$this->configHelper->credentialsAreConfigured()) {
+        if (!$this->algoliaCredentialsManager->checkCredentialsWithSearchOnlyAPIKey()) {
```

Comment:
With multi app now supported I'm wondering if we should include this in the store ID loop?

Reply:
We could consider adding the logic a little bit below (line 69), but I'm wondering why there are two separate foreach ? Do you need to finish all the deletes before starting the syncs ?

Reply:
Good question. My reasons here were two fold, 
1. I was trying to avoid coupling the logic in the `ReplicaManager` class to keep things reusable. 
2. I found that it was possible for replica misconfigurations to cross pollinate across store views which would prevent a surgical "one store at a time" approach to effectively rebuild things. Better to blow it all away first. 

As far as the validation - I think just putting it in the first loop would be sufficient?

Reply:
I've decided to check the two loops, but display the message only for the first one, if that's ok for you ? https://github.com/algolia/algoliasearch-magento-2/pull/1598/commits/7bef4ccbe69d600bb9352237c021a83bff8510b9

Reply:
Yes I'm good with that guard clause style and I think the way you've now handled this scenario is a little more graceful and works better for multi app. Just one small tweak to the language in the log message is suggested as we are not aborting the patch but just moving on to the next store.